modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchIndexSearcher.java

```diff
@@ -568,6 +568,8 @@ private Hits _searchWithDeepPagination(
 		int maxResultWindow = 10000; // GL: should be configurable
 
 		try {
+			searchSearchRequest.setStart(start);
+
 			if (end > maxResultWindow) {
```

Comment:
I  find this block difficult to read. Could it be extracted to something like:
```
			if (end > maxResultWindow) {
			
                                 searchSearchRequest.setSearchAfter(
				          _getSearchAfter(end, maxResultWindow, query, searchContext, 
						searchRequest, 
						searchSearchRequest,
						searchResponseBuilder, start));			

				searchSearchRequest.setStart(0);
			} else {

				searchSearchRequest.setStart(start);
			}
			
			searchSearchRequest.setSize(end - start);

			searchSearchResponse = _searchEngineAdapter.execute(
				searchSearchRequest);
```

Reply:
If the last SearchHit is null, is there reason not just to throw an exception?

Reply:
yep

Reply:
Actually, we can't do that because we  do check if lastSearchHit is null, if is null we are returning a new HitsImpl()

`if (lastSearchHit == null) { return new HitsImpl(); }`

so won't allow us to use else:

https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/check/java_else_statement_check.markdown#javaelsestatementcheck